### PR TITLE
gh-120221: Deliver real singals on Ctrl-C and Ctrl-Z in the new REPL

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-120221

### CODE REVIEW

**Comment by pablogsal** [comment]
@lysnikolaou Unfortunately I have not found a proper way to test this since it requires a terminal that sends something on Ctrl-C (sending signals won't test that we are not intercepting ourselves Ctrl-C neither will do it writing manually Ctrl-C to the REPL because that will never arrive if this is done properly), so this needs to be manually tested. Do you mind confirming that it works on your side?

**Comment by pablogsal** [comment]
Check https://github.com/python/cpython/issues/120221#issue-2340259901 for a reproducer

File: Misc/NEWS.d/next/Core and Builtins/2024-06-11-17-56-12.gh-issue-120221.si9hM9.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Deliver real singals on Ctrl-C and Ctrl-Z in the new REPL. Patch by Pablo
```
Comment (devdanzin) [inline Misc/NEWS.d/next/Core and Builtins/2024-06-11-17-56-12.gh-issue-120221.si9hM9.rst]:
```suggestion
Deliver real signals on Ctrl-C and Ctrl-Z in the new REPL. Patch by Pablo
```

**Comment by lysnikolaou** [comment]
This works for me! Only weird thing is that it adds some kind of weird indentation when pressing Ctrl-C more times.

```python
❯ ./python.exe
Python 3.14.0a0 (heads/paste:4e170a4361b, Jun 11 2024, 19:04:57) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import signal
>>> signal.signal(signal.SIGINT,\040lambda\040*x:\040print("NOOOO"))KeyboardInterrupt
>>> signal.signal(signal.SIGINT, lambda *x: print("NOOOO"))
<built-in function default_int_handler>
>>> NOOOO
         NOOOO
              NOOOO
                   NOOOO
                        NOOOO
                             NOOOO
```

**Comment by pablogsal** [comment]
> This works for me! Only weird thing is that it adds some kind of weird indentation when pressing Ctrl-C more times.

Yeah, but I think is because it's printing in the middle of REPL code

**Pull request merged**